makeflow/src/makeflow.c

```diff
@@ -134,29 +112,42 @@ static int file_creation_patience_wait_time = 0;
  * once weaver/pbui tools are updated.) */
 static int log_verbose_mode = 0;
 
-static char *wrapper_command = 0;
-static struct list *wrapper_input_files = 0;
-static struct list *wrapper_output_files = 0;
-
-static void makeflow_wrapper_add_command( const char *cmd )
+/* Wrapper control functions for wrapping command and adding input/output
+ *  * files. */
+static struct makeflow_wrapper *wrapper = 0;
+static struct makeflow_monitor *monitor = 0;
+  
+/* Generates file list for node based on node files, wrapper
+ *  * input files, and monitor input files. Relies on %% nodeid
+ *   * replacement for monitor file names. */
+static struct list *makeflow_generate_input_files( struct dag_node *n, struct makeflow_wrapper *w, struct makeflow_monitor *m )
 {
-	if(!wrapper_command) {
-		wrapper_command = strdup(cmd);
-	} else {
-		wrapper_command = string_wrap_command(wrapper_command,cmd);
-	}
-}
+    struct list *result = list_duplicate(n->source_files);
```

Comment:
@btovar I was creating a duplicate because I was adding to the list and we don't want the original list on the node to be modified. What would be preferable to duplicate?